node/core/provisioner/src/lib.rs

```diff
@@ -712,7 +714,20 @@ async fn select_disputes(
 	// In case of an overload condition, we limit ourselves to active disputes, and fill up to the
 	// upper bound of disputes to pass to wasm `fn create_inherent_data`.
 	// If the active ones are already exceeding the bounds, randomly select a subset.
-	let recent = request_disputes(sender, RequestType::Recent).await;
+	let mut recent = request_disputes(sender, RequestType::Recent).await;
```

Comment:
Doesn't the dispute coordinator system also scrape onchain votes? I imagine we can move this there and then we could have a `RequestType::RecentOffChain` (not sure if this is the best naming) request type that we can use here.

Reply:
If I'm not mistaken what's scraped in the disputes-coordinator are the resolved disputes while here we are fetching the ongoing disputes so one way or another - we'll have to make the runtime API call.

In which module to do it - you have a point. It can also be done in the disputes-coordinator but for me it felt more natural to keep this logic in the provisioner because the inherent data is prepared there. I don't insist on this approach though. Keeping it in the disputes-coordinator is very reasonable too.

Reply:
The on chain votes are the votes that are imported in that very block, not the overall disputes. Put another way, the votes the provisioner provides can later be found in the on chain votes of the created block. This is a backup mechanism, in case a validator misses disputes off-chain, it will still know about current disputes via those onchain vote scraping.